pkg/plugin/types.go

```diff
@@ -11,10 +11,18 @@ import (
 )
 
 const (
-	FuncName = "Process"
+	ProcessFuncName     = "Process"
+	IsSupportedFuncName = "IsSupported"
 )
 
-type Func func(resource smith_v1.Resource, dependencies map[smith_v1.ResourceName]Dependency) (ProcessResult, error)
+type Process func(resource smith_v1.Resource, dependencies map[smith_v1.ResourceName]Dependency) (ProcessResult, error)
+type IsSupported func(plugin string) (bool, error)
+
+type Plugin struct {
+	Name        string
+	Process     Process
+	IsSupported IsSupported
```

Comment:
So a single plugin can support multiple "Bundle plugin names"?
e.g. a single shim plugin can proxy calls to all "actual" plugin processes?
This looks fine, just not sure if using the "plugin" term in both context is correct there, it's a bit confusing.

The function can probably be called something like `Implements(...)`

Reply:
`Implements` is a bad name too I think :)